src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -1050,5 +1050,123 @@ public static T WithCommentsFrom<T>(this T node, SyntaxToken leadingToken, Synta
             => node
             .WithTrailingTrivia(node.GetTrailingTrivia().FilterComments(addElasticMarker: true))
             .WithLeadingTrivia(node.GetLeadingTrivia().FilterComments(addElasticMarker: true));
+
+        // Once https://github.com/dotnet/roslyn/issues/30116 is implemented, we can switch this API
+        // and other extension methods in this file that use this API to be language-agnostic extensions.
+        // TODO: File a tracking a track bug for this API cleanup.
```

Comment:
unlike below, this is minor enough that i'm ok with us going in even before 30116 happening.

Reply:
However, thinking about it more... it's definitely a bit worrying to me.  The problem here is that people have to know that depending on if  you go node->op->value versus node->value, you can get different results... seems easy to make mistakes.